lib/CustomFields/pages/EditCustomFieldsSettings/EditCustomFieldsSettings.js

```diff
@@ -173,25 +173,36 @@ const EditCustomFieldsSettings = ({
     }));
   };
 
-  const saveCustomFields = async (data, redirectToView = true) => {
+  const saveCustomFields = async (data, redirectToView = true, titleShouldUpdate = true) => {
     setSubmitting(true);
 
-    Promise.all([
-      updateCustomFields(formatCustomFieldsForBackend(data)),
-      updateTitle(data.sectionTitle),
-    ]).then(responses => {
-      if (responses.every(({ ok }) => !!ok)) {
-        if (redirectToView) {
-          history.push(viewRoute);
-        }
+    if (titleShouldUpdate) {
```

Comment:
For me it feels like there are too many conditions and it's hard to understand what's going on here. Plus, there's a lot of duplicated code inside both `if` and `else`
I'd suggest to first build an array of requests, and then use `Promise.all`.

```suggestion
    const requests = [updateCustomFields(formatCustomFieldsForBackend(data)];
    
    if (titleShouldUpdate) {
      requests.push(updateTitle(data.sectionTitle);
    }
    
    Promise.all(requests).then(responses => {
      if (responses.every(({ ok }) => !!ok)) {
        if (titleShouldUpdate && redirectToView) {
          history.push(viewRoute);
        }
        
        setCustomFields(formatCustomFieldsForSaving(data));
      } else {
        showErrorNotification();
        setSubmitting(false);
      }
    });
```

Reply:
I did not even think of such a decision. You're right, your suggestion is much better. Thanks.